ldap_auth_provider.py

```diff
@@ -321,9 +325,11 @@ class _LdapConfig(object):
         ldap_config.start_tls = config.get("start_tls", False)
         ldap_config.base = config["base"]
         ldap_config.attributes = config["attributes"]
```

Comment:
these seem to be unused when a `bind_pattern` is specified. Can you modify the `_requre_keys` calls to make them optional?